pkg/agent/controller/egress/egress_controller.go

```diff
@@ -209,6 +219,39 @@ func (c *EgressController) onLocalIPUpdate(ip string, added bool) {
 	}
 }
 
+func (c *EgressController) onClusterNodeUpdate(nodeName string, join bool) {
+	var egresses []*crdv1a2.Egress
+	if join {
+		//	list all egress of local node, and move not hit egress
+		c.egressStatesMutex.RLock()
+		defer c.egressStatesMutex.RUnlock()
+		for egressName := range c.egressStates {
+			egress, _ := c.egressLister.Get(egressName)
+			if egress.Spec.ExternalIPPool != "" && c.cluster.ShouldSelectEgress(egress) {
+				continue
+			}
+			egresses = append(egresses, egress)
+		}
+		klog.InfoS("Detected cluster node joined, egress should unbind", "nodeName", nodeName)
+	} else {
+		//	list egress owned by left node and handler the egress if hit by local node
+		egressesOfLeftNode, _ := c.egressLister.List(labels.Everything())
```

Comment:
The variable nevers points to egresses of left node. It should just be `egresses`.

Reply:
done